src/main/java/com/ryszardpanda/medicalClinic/PatientRepository.java

```diff
@@ -0,0 +1,48 @@
+package com.ryszardpanda.medicalClinic;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+@RequiredArgsConstructor
+public class PatientRepository {
+
+    private final List<Patient> patients;
```

Comment:
To co robi Spring w sytuacji jak wstrzykujesz Listę czegokolwiek to on wtedy przeszukuje wszystkie zdefiniowane beany dla danego typu i wstrzykuje wsystkie znalezione beany. Przykładowo jest to przydatne w sytuacji jak mamy np. jeden interfejs i 5 jego implementacji.

Reply:
W twojej sytuacji nie znajdzie beana dla Paitentów wiec lista bedzie pusta na poczatku.